Composer/plugins/runtimes/src/index.ts

```diff
@@ -0,0 +1,220 @@
+        throw err;
+        return;
+      }
+      // Then, copy the declarative assets into the build artifacts folder.
+      const remoteBotPath = path.join(publishFolder, 'ComposerDialogs');
+      const localBotPath = path.join(runtimePath, 'ComposerDialogs');
+      await fs.copy(localBotPath, remoteBotPath, {
+        overwrite: true,
+        recursive: true,
+      });
+
+      // write settings to disk in the appropriate location
+      const settingsPath = path.join(publishFolder, 'ComposerDialogs', 'settings', 'appsettings.json');
+      if (!(await fs.pathExists(path.dirname(settingsPath)))) {
+        fs.mkdirSync(path.dirname(settingsPath), { recursive: true });
+      }
+      fs.writeFileSync(settingsPath, JSON.stringify(settings, null, 2));
+
+      // return the location of the build artifiacts
+      return publishFolder;
+    },
+    eject: async (project, localDisk: IFileStorage) => {
+      const sourcePath = path.resolve(__dirname, '../../../../runtime/dotnet');
+      const destPath = path.join(project.dir, 'runtime');
+      if (!(await project.fileStorage.exists(destPath))) {
+        // used to read bot project template from source (bundled in plugin)
+        await copyDir(sourcePath, localDisk, destPath, project.fileStorage);
+        const schemaDstPath = path.join(project.dir, 'schemas');
+        const schemaSrcPath = path.join(sourcePath, 'azurewebapp/schemas');
+        const customSchemaExists = fs.existsSync(schemaDstPath);
+        const pathsToExclude: Set<string> = new Set();
+        if (customSchemaExists) {
+          const sdkExcludePath = await localDisk.glob('sdk.schema', schemaSrcPath);
+          if (sdkExcludePath.length > 0) {
+            pathsToExclude.add(path.join(schemaSrcPath, sdkExcludePath[0]));
+          }
+        }
+        await copyDir(schemaSrcPath, localDisk, schemaDstPath, project.fileStorage, pathsToExclude);
+        const schemaFolderInRuntime = path.join(destPath, 'azurewebapp/schemas');
+        await removeDirAndFiles(schemaFolderInRuntime);
+        return destPath;
+      }
+      throw new Error(`Runtime already exists at ${destPath}`);
+    },
+    setSkillManifest: async (
+      dstRuntimePath: string,
+      dstStorage: IFileStorage,
+      srcManifestDir: string,
+      srcStorage: IFileStorage,
+      mode = 'azurewebapp' // set default as azurewebapp
+    ) => {
+      // update manifst into runtime wwwroot
+      if (mode === 'azurewebapp') {
+        const manifestDstDir = path.resolve(dstRuntimePath, 'azurewebapp', 'wwwroot', 'manifests');
+
+        if (await fs.pathExists(manifestDstDir)) {
+          await removeDirAndFiles(manifestDstDir);
+        }
+
+        if (await fs.pathExists(srcManifestDir)) {
+          copyDir(srcManifestDir, srcStorage, manifestDstDir, dstStorage);
+        }
+      }
+    },
+  });
+
+  composer.addRuntimeTemplate({
+    key: 'node-azurewebapp',
+    name: 'JS (preview)',
+    startCommand: 'node ./lib/webapp.js',
```

Comment:
Update start command to `npm start`

Reply:
if we use npm start instead, it may cause issue when restart. Because the process about js runtime is more than one, and could not be kill completely to reuse the port.